src/aws_credentials_ec2.erl

```diff
@@ -3,40 +3,70 @@
 -module(aws_credentials_ec2).
 -behaviour(aws_credentials_provider).
 
+-define(SESSION_TOKEN_URL,
+        "http://169.254.169.254/latest/api/token/").
+-define(SESSION_TOKEN_TTL_HEADER,
+        "x-aws-ec2-metadata-token-ttl-seconds").
+-define(SESSION_TOKEN_TTL_SECONDS,
+        "21600").
+-define(SESSION_TOKEN_HEADER,
+        "X-aws-ec2-metadata-token").
 -define(CREDENTIAL_URL,
         "http://169.254.169.254/latest/meta-data/iam/security-credentials/").
 -define(DOCUMENT_URL,
         "http://169.254.169.254/latest/dynamic/instance-identity/document").
 
 -type role() :: binary().
+-type session_token() :: binary().
```

Comment:
I'm new to Erlang and feel like I'm possibly spec'ing too much?  🤷🏻  Please educate me.

Reply:
I think you're doing great. This kind of "aliasing" for types improves readability in my opinion.